### PR TITLE
Fix Managed identity 

### PR DESCRIPTION
**Pull request by coin8086**

### CODE REVIEW

File: specs/default/chef/site-cookbooks/hpcpack/files/InstallUtilities.psm1 [hunk 1]
```diff
@@ -111,12 +112,16 @@ function Get-MsiAccessToken
...
+        [string] $ManagedId
     )
-    
+
     $Resource = $Resource.Trim()
     $encodedResource = [uri]::EscapeDataString($Resource)
-    $tokenUri = "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=$encodedResource"
+    $encodedManagedId = [uri]::EscapeDataString($ManagedId)
+    $tokenUri = "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=$encodedResource&msi_res_id=$encodedManagedId"
     $resp = Invoke-WebRequest -Uri $tokenUri -Method Get -Headers @{Metadata="true"} -UseBasicParsing
```
Comment (sunbinzhu) [inline specs/default/chef/site-cookbooks/hpcpack/files/InstallUtilities.psm1:125]: shall we explicitly specify 'No proxy' option here?

File: templates/hpcpack.txt [hunk 2]
```diff
@@ -20,6 +20,7 @@ Autoscale = $Autoscale
       cyclecloud.cluster.autoscale.provisioning_timeout = 1500
       cyclecloud.cluster.autoscale.vm_retention_days = $VMRetentionDays
       cyclecloud.node.prevent_metadata_access = false
+      hpcpack.identity = $ManagedServiceIdentity
```
Comment (sunbinzhu) [inline templates/hpcpack.txt]: "identity" sounds too general.

File: specs/default/chef/site-cookbooks/hpcpack/recipes/_get_secrets.rb [hunk 3]
```diff
@@ -26,46 +26,31 @@
...
 end
 
+Chef::Log.info( "The managed identity is `#{node[:hpcpack][:identity]}'." )
+
 # Lookup the AD Admin and Cert creds in KeyVault (if present)
 if ! node['hpcpack']['keyvault']['vault_name'].nil?
   Chef::Log.info( "Looking up secrets in vault: #{node['hpcpack']['keyvault']['vault_name']}..." )
 
-
-  if ! node['hpcpack']['keyvault']['admin']['name_key'].nil?
```
Comment (sunbinzhu) [inline specs/default/chef/site-cookbooks/hpcpack/recipes/_get_secrets.rb:34]: I suppose admin user name can be passed as plaintext. Do we need to store it in key vault as secret?

File: templates/hpcpack.txt (hunk 2 above)
Comment (sunbinzhu) [inline templates/hpcpack.txt]: Do we need to consider the backward compatibility where this parameter is not defined in clusters deployed with previous version?

File: specs/default/chef/site-cookbooks/hpcpack/recipes/_get_secrets.rb (hunk 3 above)
Comment (coin8086) [reply specs/default/chef/site-cookbooks/hpcpack/recipes/_get_secrets.rb:34]: It's not stored as a secret, so I deleted it.

File: templates/hpcpack.txt (hunk 2 above)
Comment (coin8086) [reply templates/hpcpack.txt]: So what about "hpcpack.managedIdentity"?

File: specs/default/chef/site-cookbooks/hpcpack/files/InstallUtilities.psm1 (hunk 1 above)
Comment (coin8086) [reply specs/default/chef/site-cookbooks/hpcpack/files/InstallUtilities.psm1:125]: Sure

**Review by bwatrous — APPROVED** [review]

File: templates/hpcpack.txt (hunk 2 above)
Comment (coin8086) [reply templates/hpcpack.txt]: By Ben's reply in email, we don't need to worry about it.

**Pull request merged**